File: UCA-4/Arrays/C/swap.c
//Swapping Two Variables Value
#include <stdio.h>

void swap(int *n1,int *n2){
	*n1 = *n1 + *n2;
	*n2 = *n1 - *n2;
	*n1 = *n1 - *n2;
}

int main(int argc, char const *argv[])
{
	int num1 = atoi(argv[1]);
	int num2 = atoi(argv[2]);
	printf("X: %d Y: %d\n",num1,num2);
	swap(&num1,&num2);
	printf("X: %d Y: %d\n",num1,num2);
	return 0;
}

File: UCA-4/Arrays/Java/swap.java
//Swapping Two Variables Value
/* IMP
* In Java Methods use PASS-BY-VALUE instead of PASS-BY-REFERENCE
* Therefore used this trick to swap value
*/
import java.util.*;

class Test{
	static int swap(int num1,int num2){
		return num1;
	}

	public static void main(String[] args) {
		if(args!=null){
			int num1 = Integer.parseInt(args[0]);	
			int num2 = Integer.parseInt(args[1]);
			System.out.println("X: "+num1+" Y: "+num2);
			num1 = swap(num2,num2=num1);
			System.out.println("X: "+num1+" Y: "+num2);
		}
		
	}
}
